lib/puppet/module/task.rb

```diff
@@ -61,6 +62,29 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.find_files(name, metadata)
+      if metadata.key?('files')
+        file_list = metadata['files'].map do |file|
+          splits = file.split(File::SEPARATOR)
+          pup_module = Puppet::Module.find(splits[0])
+          if pup_module.nil?
+            warning _("Could not find module %{module}" % { module: splits[0] })
+            return []
+          end
+          mount = splits[1]
+          unless MOUNTS.include? mount
+            msg = _("Files must be saved in module directories that Puppet makes available via mount points: files, lib, tasks")
+            raise InvalidMetadata.new(msg, 'puppet.tasks/invalid-metadata')
+          end
+          path = File.join(pup_module.path, mount, splits[2..-1])
```

Comment:
This part we should make sure attempting to read a file outside the module. We can use a test like `File.absolute_path(path) == File.path(path)`.

Reply:
Actually, Puppet appears to have several different implementations of this idea, generally named `absolute?`. `Pathname.new(path).absolute?` looks like the most straightforward one.

Reply:
Added an absolute path check below!